ovirt_imageio/client/_options.py

```diff
@@ -45,13 +45,34 @@ def __repr__(self):
 log_level = Choices("log_level", ("debug", "info", "warning", "error"))
 
 
+class StoreFalseOrNoneAction(argparse.Action):
+    """
+    This action stores 'False' value, similar to 'store_false' but without
+    the implicit default 'True' value.
+    This action is to be used with optional arguments that specify
+    some sort of flag.
+    """
+
+    def __init__(self, option_strings, dest, required=False, help=None):
+        super().__init__(
+            option_strings=option_strings,
+            dest=dest,
+            nargs=0,
+            required=required,
+            help=help)
```

Comment:
If we just pass arguments to the super constructor, we usually don't need to implement
`__init__`.

Looks like this version passes option_strings, dest, required, and help, and use nargs=0 as default. Don't we get the same result if we don't implement `__init__`?

Reply:
By default, `nargs` takes a single value, but we want it to explicitly be `0`.

Reply:
We can simplify this a lot like this:

```
$ cat opt.py 
import argparse

p = argparse.ArgumentParser()
p.add_argument("--insecure", dest="secure", action='store_false', default=None)
args = p.parse_args()
print(args)

$ python opt.py --insecure
Namespace(secure=False)

$ python opt.py
Namespace(secure=None)
```

So the only thing to do to make this work is to pass

    default=None

when using store_xxx actions, because these action classes uses:

```
class _StoreFalseAction(_StoreConstAction):

    def __init__(self,
                 option_strings,
                 dest,
                 default=True,
                 required=False,
                 help=None):
        super(_StoreFalseAction, self).__init__(
            option_strings=option_strings,
            dest=dest,
            const=False,
            default=default,
            required=required,
            help=help)
```

Since we never want to set the default using argparse, we we need to add

    "default": None

To our new shiny kwargs() property.

Reply:
I realized the `default=True` in the store Action, but didn't think about leveraging the `kwargs` option to default it to `None` and then set it later. That's a nice trick!